pkg/handler/enqueue_owner.go

```diff
@@ -197,3 +201,49 @@ func (e *enqueueRequestForOwner) getOwnersReferences(object metav1.Object) []met
 	// No Controller OwnerReference found
 	return nil
 }
+
+// logEvent logs an event with the version, kind and name of the object and its owner.
+func (e *enqueueRequestForOwner) logEvent(eventType string, object, newObject client.Object) {
+	var ownerReference *metav1.OwnerReference
+	if e.ownerType != nil && object != nil {
+		ownerReference = extractTypedOwnerReference(e.ownerType.GetObjectKind().GroupVersionKind(), object.GetOwnerReferences())
+		if ownerReference == nil && newObject != nil {
+			ownerReference = extractTypedOwnerReference(e.ownerType.GetObjectKind().GroupVersionKind(), newObject.GetOwnerReferences())
+		}
+	}
+
+	// If no ownerReference was found then it's probably not an event we care about
+	if ownerReference != nil {
+		kvs := []interface{}{
+			"Event type", eventType,
+			"GroupVersionKind", object.GetObjectKind().GroupVersionKind().String(),
+			"Name", object.GetName(),
+		}
+		if objectNs := object.GetNamespace(); objectNs != "" {
+			kvs = append(kvs, "Namespace", objectNs)
+		}
+		kvs = append(kvs,
+			"Owner APIVersion", ownerReference.APIVersion,
+			"Owner Kind", ownerReference.Kind,
+			"Owner Name", ownerReference.Name,
+		)
+
+		log.Info("OwnerReference handler event", kvs...)
```

Comment:
What "level" does this get logged at? Is there a way for me to specify that I only want this logged at a certain verbosity level?

Reply:
```suggestion
		log.V(1).Info("OwnerReference handler event", kvs...)
```

I don't think this should be logged at the normal Info level, as it will create too chatty logs. See https://github.com/kubernetes-sigs/controller-runtime/blob/main/TMP-LOGGING.md

Reply:
Yeah, this is defininitely debug level, it will cause a huge amount of logs for a controller that is even slightly busy